### ISSUE TITLE
Auto height based on aspect ratio

### ISSUE THREAD

**Issue by xcambar** [videojs/video.js#771]
Hi,
Hers is a description of what I'm trying to achieve:
- videos have a flexible width based upon the dimensions of the container.
  - => That's easy with `width: auto`
- the height of a video is automatic based upon the aspect ratio and the width (pretty much like a native tag would do)
  - No idea how to do it

I've been playing with various combinations of `width` and `height`, be it `auto` or `100%`, and could not get a good result

Is there an option I could set to achieve this or a method I could call to know the dimensions of the video source, so I can compute myself the correct height?
### Example

By writing something like this:

``` html
<div class="container">
  <video
    data-setup='{}'
    id="123"
    class="video-js vjs-default-skin"
    controls
    height="auto"
    width="auto"
    preload="auto">
    <source src="http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4" type="video/mp4"></source>
  </video>
</div>
```

If my video is `800*600` and `.container` has a width of `400px`, the height of the videojs 'component' should be `300px`.

Is this something that can be achieved ?

**Note**: This may be related to #359

**Comment by SimonEast** [comment]
Has there been any progress on this since Feb 2014?

**Comment by mmcc** [comment]
@SimonEast Yep! Thanks for the reminder, this should be closed now that #1952 is pulled into master.
